src/ok.rs

```diff
@@ -605,3 +671,209 @@ fn memlock_rlimit() -> TestResult {
         mesg: Some(mesg),
     }
 }
+
+fn msr_bit_read(reg:u32, cpu:u16) -> Result<u64,anyhow::Error> {
+    let mut msr = Msr::new(reg, cpu).context("Error Reading MSR")?;
+    let raw_value = msr.read()?;
+    Ok(raw_value)
+}
+
+fn sme_test() -> TestResult{
+        let raw_value = match msr_bit_read(0xC0010010,0) {
+            Ok(raw) => raw,
+            Err(e) => return TestResult{
+                        name: "SME Test".to_string(),
+                        stat: TestState::Fail,
+                        mesg: format!("MSR read failed: {}", e).into()
+            }
+        };
+        let (testres,mesg) = match raw_value >> 23 & 1{
+            1 => (TestState::Pass, "Enabled"),
+            0 => (TestState::Fail, "Disabled"),
+            _ => (TestState::Fail, "Invalid Number Found")
```

Comment:
You might just do `unreachable!()`, as this is checking a single bit. @DGonzalezVillal @tylerfanelli thoughts about that?

Reply:
@larrydewey Sure, are you talking about when I'm looking for invalid numbers?